rpc_util.go

```diff
@@ -499,6 +549,28 @@ func getMaxSize(mcMax, doptMax *int, defaultVal int) *int {
 	return doptMax
 }
 
+// should be called after options applied
+func setCallInfoContentSubtypeAndCodec(c *callInfo) error {
+	if c.contentSubtype != "" {
+		if c.codec == nil {
+			// c.contentSubtype is already lowercased in CallContentSubtype
+			codec, ok := registeredCodecs[c.contentSubtype]
+			if !ok {
+				return status.Errorf(codes.Internal, "no codec registered for content-subtype %s", c.contentSubtype)
+			}
+			c.codec = codec
+		}
+	} else {
+		if c.codec != nil {
```

Comment:
How about this:

```go
if c.codec == nil {
  c.codec = encoding.GetDefaultCodec()
}
c.contentSubtype = strings.ToLower(c.codec.String())
```

Reply:
Same discussion regarding default codec - there's already ways to set default codecs, this way is also backwards-compatible in assuming no codec means the proto codec.